cmake/developer_package/features.cmake

```diff
@@ -49,6 +49,8 @@ ov_dependent_option (ENABLE_AVX2 "Enable AVX2 optimizations" ON "X86_64 OR (X86
 
 ov_dependent_option (ENABLE_AVX512F "Enable AVX512 optimizations" ON "X86_64 OR (X86 AND NOT EMSCRIPTEN)" OFF)
 
+ov_dependent_option(ENABLE_ARM_FP16 "Enable ARM FP16 optimizations" ON "AARCH64 OR (ARM AND NOT EMSCRIPTEN)" OFF)
```

Comment:
Let's keep the condition as `AARCH64` for now
Definetely `EMSCRIPTEN` is not required here